Add autogenerated OPAM file

craigfe opened the pull request:
Generated using `dune-project` metadata.

This is necessary to install the appropriate dependencies in CI.

craigfe committed:
Add autogenerated OPAM file

craigfe committed:
Reformat with OCamlformat

talex5 commented on dune-project:
```diff
@@ -1,2 +1,12 @@
+(package
+ (name citty)
+ (synopsis "CI in TTY")
+ (depends lwt notty fpath mirage-flow-unix))
```
Why does it depend on `mirage-flow-unix`? I guess this should be `capnp-rpc-unix` (looking at `src/dune`).

craigfe replied:
I copied over these dependencies from the README: https://github.com/ocaml-ci/citty#2-install-dependencies. I think it's not necessary to depend directly on `capnp-rpc-unix`, since this repository contains it as a submodule?

talex5 replied:
Well, it won't contain it if someone updates to the latest version of ocaml-ci, and it won't work if someone tries to publish citty to opam.

craigfe replied:
Good points; I'll fix the .opam file and the README accordingly

craigfe commented:
I have updated the `.opam` file to include dependencies in submodules, and added an instruction in the README to pin those dependencies before installing from the `.opam` file.

craigfe commented:
`capnp-rpc-lwt.0.5.0` now fails to install, perhaps because it needs to be a `pin-depends` instead? Will take a look later.

talex5 commented:
I think it just needs updating to the latest ocaml-ci. The version of ocaml-ci used in this PR vendors an old copy of capnp-rpc, which is missing an upper bound on x.509 (which broke its API a while back).

craigfe committed:
Update ocaml-ci submodule

craigfe commented:
Agreed, done. Thanks :slightly_smiling_face:

craigfe commented:
Rebased and re-updated the `ocaml-ci` version.
